linux/LinuxProcessList.c

```diff
@@ -1971,6 +1985,7 @@ void ProcessList_goThroughEntries(ProcessList* super, bool pauseProcessUpdate) {
    LinuxProcessList_scanMemoryInfo(super);
    LinuxProcessList_scanHugePages(this);
    LinuxProcessList_scanZfsArcstats(this);
+   LinuxProcessList_updateTime(this);
```

Comment:
Maybe insert this function call as first action, so every function can assume the data is up-to-date.

Reply:
There's still a separate gettimeofday call in checkRecalculation ScreenManager.c too - isn't that the very first place we could capture the 'current time of sampling' (perhaps via a new routine named like Platform_getCurrentTime?)  Then we'd not need a later LinuxProcessList_updateTime call.

Reply:
@natoscott Excellent, I can't call Platform_getCurrentTime directly from ScreenManager.c (this causes problems with CI and other OSs)
So, I made ProcessList_getCurrentTime.
I think this's optimal because our time-related variables exist within ProcessList struct (ProcessList.h)  and my new function in ProcessList.c. I could call it directly from there also.

Reply:
@smalinux I'll send some code through shortly - we should be able to call platform code from ScreenManager.c if we need to.  I'll need to reproduce that CI failure though to see the exact issue.